src/db/models/Notifications.js

```diff
@@ -0,0 +1,154 @@
+import mongoose from 'mongoose';
+import Random from 'meteor-random';
+import { MODULES } from '../../data/constants';
+
+// Notification schema
+const NotificationSchema = new mongoose.Schema({
+  _id: {
+    type: String,
+    unique: true,
+    default: () => Random.id(),
+  },
+  notifType: {
+    type: String,
+    enum: MODULES.ALL,
+  },
+  title: String,
+  link: String,
+  content: String,
+  createdUser: String,
+  receiver: String,
+  date: {
+    type: Date,
+    default: Date.now,
+  },
+  isRead: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+class Notification {
+  /**
+   * Marks notifications as read
+   * @param {String[]} ids - Notification ids
+   * @return {Promise}
+   */
+  static markAsRead(ids) {
+    return this.update({ _id: { $in: ids } }, { $set: { isRead: true } }, { multi: true });
+  }
+
+  /**
+   * Create a notification
+   * @param {Object} doc - Notification object
+   * @param {string} doc.notifType - Category of notification  (module)
+   * @param {string} doc.title - Notificaton title
+   * @param {string} doc.content - Notification content
+   * @param {string} doc.link - Notification link
+   * @param {Object|string} doc.receiver - Id of the user that will receive this notification
+   * @param {Object|string} createdUser - The user whose actions made this notification
+   * @return {Notification} Notification Object
+   * @throws {Exception} throws Exception if createdUser is not supplied
+   */
+  static async createNotification(doc, createdUser) {
+    if (!createdUser) {
+      throw new Error('createdUser must be supplied');
+    }
+
+    // Setting auto values
+    doc.createdUser = createdUser;
+
+    // if receiver is configured to get this notification
+    const config = await NotificationConfigurations.findOne({
+      user: doc.receiver,
+      notifType: doc.notifType,
+    });
+
+    // receiver disabled this notification
+    if (config && !config.isAllowed) {
+      throw new Error('Configuration does not exist');
+    }
+
+    return await this.create(doc);
+  }
+
+  /**
+   * Update a notification
+   * @param {string} _id - Id of notification
+   * @param {Object} doc - Notification object
+   * @param {string} doc.notifType - Category of notification  (module)
+   * @param {string} doc.createdUser - The user whose actions made this notification
+   * @param {string} doc.title - Notificaton title
+   * @param {string} doc.content - Notification content
+   * @param {string} doc.link - Notification link
+   * @param {Object|string} doc.receiver - Id of the user that will receive this notification
+   * @return {Promise} The promise returns null
+   */
+  static updateNotification(_id, doc) {
+    return this.update({ _id }, doc);
```

Comment:
return updated notification
await this.update ....
return this.findOne({ _id })